apps/core/scenes/results/ResultsListItem.js

```diff
@@ -1,61 +1,246 @@
+      const stopAfterSector = route[sectorBorderIndex];
+
+      const utcDeparture =
+        stopAfterSector &&
+        stopAfterSector.departure &&
+        DateTime.fromISO(stopAfterSector.departure.utcTime);
+      const utcArrival =
+        stopBeforeSector &&
+        stopBeforeSector.arrival &&
+        DateTime.fromISO(stopBeforeSector.arrival.utcTime);
+      return this.getDateTimeInterval(utcDeparture, utcArrival, [
+        'days',
+        'hours',
+      ]);
+    }
     return null;
-  }
-  const { flyFrom, flyTo, localDeparture, localArrival, price } = props.data;
-  /**
-   * TODO: properly handle possible undefined props in final version of list
-   * (for example: should be entry with undefined price still displayed?)
-   *
-   * NOTE: ugly form of string creation below is used for now because
-   * using ?? inside template literals caused problems with code styling
-   * inside VS Code (this part will be probably completely replaced
-   * in next iteration, so it's temporary solution)
-   */
-  const itemString =
-    (flyFrom ?? 'Unknown destination') +
-    ' / ' +
-    (localDeparture ?? 'Unknown departure') +
-    ' -> ' +
-    (flyTo ?? 'Unknown destination') +
-    ' / ' +
-    (localArrival ?? 'Unknown arrival') +
-    ' | ' +
-    (price ?? 'Unknown price');
-  return <Text style={styles.listItem}>{itemString}</Text>;
-};
+  };
 
-const styles = StyleSheet.create({
-  listItem: {
-    fontSize: 12,
-    padding: 2,
-    margin: 2,
-    backgroundColor: defaultTokens.backgroundBody,
-  },
-});
+  render() {
+    const { data } = this.props;
+    const sanitizedRoute = this.getSanitizedRoute();
+    const sectorBorderIndex = this.getSectorBorderIndex();
+    const hasSectorBorder = sectorBorderIndex != null && sectorBorderIndex > -1;
+    if (data == null) {
+      return null;
+    }
+    const { price } = data;
+    /**
+     * TODO: properly handle possible undefined props in final version of list
+     * (for example: should be entry with undefined price still displayed?)
+     *
+     * NOTE: ugly form of string creation below is used for now because
+     * using ?? inside template literals caused problems with code styling
+     * inside VS Code (this part will be probably completely replaced
+     * in next iteration, so it's temporary solution)
+     */
+    const badges = [
+      {
+        id: 1,
+        type: 'warning',
+        children: 'Cheapest',
+      },
+      {
+        id: 2,
+        type: 'neutral',
+        children: 'Wi-fi',
+      },
+    ];
+    const priceObject = {
+      amount: parseFloat(price?.amount) ?? 0,
+      currency: price?.currency ?? 'CZK',
+      locale: 'cs-CZ',
+    };
+
+    const duration =
+      this.intervalToString(
+        this.getDurationBetweenSectors(data.route, sectorBorderIndex),
+        'days',
+      ) ?? '';
+    const hasReturnFlight = sectorBorderIndex != null && sectorBorderIndex > 0;
+
+    // @TODO: use Card component
```

Comment:
Which Card component?

Reply:
It's only about styling. You can see that every connection has white background and two gray borders, so this should be `Card`.
<img width="339" alt="screenshot 2019-01-11 at 16 10 34" src="https://user-images.githubusercontent.com/858321/51041995-7525f400-15bb-11e9-92ee-909016756b83.png">